### PR TITLE
feat(Adding new component ReportChunkNames): Adds an alternative way …

### PR DESCRIPTION
**Pull request by rileylnapier**
…to report chunk names that does not require global cache and instead relies on a Provider type component and context type function to callback the chunkname.

example

```
import { ReportChunks } from 'react-universal-component'

const chunkNames = []
const appHtml = renderToString(
      <ReportChunks report={chunkName => chunkNames.push(chunkName)}>
        <App />
      </ReportChunks>
);

const { scripts } = flushChunks(clientStats, {
     chunkNames,
})
```

### CODE REVIEW

File: src/requireUniversalModule.js [hunk 1]
```diff
@@ -138,16 +138,19 @@ export default function requireUniversalModule<Props: Props>(
...
         const name = callForString(chunkName, props)
         if (name) CHUNK_NAMES.add(name)
-        if (!isTest) return // makes tests way smaller to run both kinds
+        if (!isTest) return name // makes tests way smaller to run both kinds
       }
 
       if (isWebpack()) {
         const weakId = callForString(resolve, props)
         if (weakId) MODULE_IDS.add(weakId)
+        return weakId
```
Comment (rileylnapier) [inline src/requireUniversalModule.js]: im not sure if we would need to return the weakIds for anything...  but it seemed to match the pattern of returning the name above

**Comment by rileylnapier** [comment]
let me know if this concept is ok, and i will fix the flow reports and code climate.  it works for my use case so i would *love* to see it land 👍

File: src/requireUniversalModule.js (hunk 1 above)
Comment (faceyspacey) [reply src/requireUniversalModule.js]: yea, that's fine. that's for the old approach. the only problem is the component is called `<ReportChunks>`, but it doesn't matter. I'm fine iwth it.

**Comment by faceyspacey** [comment]
This is looking very good. pretty much mergable, minus the flow types. Give them a try, they're not too hard:

https://flow.org/en/docs/react/components

You pretty much just need to do this line:
```
React.Component<Props, State>
```

**Comment by faceyspacey** [comment]
also make sure to re-export in the server.js file:

https://github.com/faceyspacey/react-universal-component/blob/master/server.js

**Comment by rileylnapier** [comment]
you think just export in server file or export in both?

**Comment by faceyspacey** [comment]
Both

**Comment by faceyspacey** [comment]
The readme formatting also needs to be fixed. Only the new paragraphs you added should show in its diff. Your editor is doing some additional formatting. For example it's using _ instead of * for italics.

**Comment by rileylnapier** [comment]
kk ill have to turn off my prettier plugin

**Comment by faceyspacey** [comment]
Also make sure to keep using npm run cm

File: README.md [hunk 2]
```diff
@@ -1,108 +1,69 @@
-<a href="https://gitter.im/Reactlandia/Lobby" target="_blank">
-  <img alt="Reactlandia Chat" src="https://s3-us-west-1.amazonaws.com/cdn.reactlandia.com/reactlandia-chat.png">
-</a>
+[![Reactlandia Chat](https://s3-us-west-1.amazonaws.com/cdn.reactlandia.com/reactlandia-chat.png)](https://gitter.im/Reactlandia/Lobby)
```
Comment (rileylnapier) [inline README.md]: blah.  let me try again on this readme format

File: README.md [hunk 3]
```diff
@@ -213,6 +213,33 @@ export default function serverRender(req, res) => {
   `)
 ```
 
+> NOTE: this requires that the bundling and rendering happen within the same context. The module, react-universal-component/server holds a global cache of all the universal components that are rendered and makes them available via `flushChunkNames`
```
Comment (rileylnapier) [inline README.md]: yay!  figured it out.  i had atom-beautify and prettier-atom both installed grrr

**Comment by faceyspacey** [comment]
Nicely done
